apps/evm_family/evm_txn_helpers.h

```diff
@@ -0,0 +1,170 @@
+/**
+ * @file    evm_txn_helpers.h
+ * @author  Cypherock X1 Team
+ * @brief   Helper apis for interpreting and signing EVM transactions
+ * @copyright Copyright (c) 2023 HODL TECH PTE LTD
+ * <br/> You may obtain a copy of license at <a href="https://mitcc.org/"
+ * target=_blank>https://mitcc.org/</a>
+ */
+#ifndef EVM_TXN_HELPERS_H
+#define EVM_TXN_HELPERS_H
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+
+#include "coin_utils.h"
+
+/*****************************************************************************
+ * MACROS AND DEFINES
+ *****************************************************************************/
+
+#define ETHEREUM_PURPOSE_INDEX 0x8000002C
+#define ETHEREUM_COIN_INDEX 0x8000003C
+
+#define ETH_VALUE_SIZE_BYTES (32U)
+
+/*****************************************************************************
+ * TYPEDEFS
+ *****************************************************************************/
+
+/**
+ * @brief Enum used to represent the status of payload field in a transaction.
+ *
+ */
+typedef enum {
+  PAYLOAD_ABSENT = 0x0,                 // No payload present in the transaction
+  PAYLOAD_SIGNATURE_NOT_WHITELISTED,    // Payload function signature is not
+                                        // recognized [Blind Signing]
+  PAYLOAD_CONTRACT_NOT_WHITELISTED,     // [OBSOLETE] Payload function signature
+                                        // is whitelisted but contract is not
+                                        // (for Transfer function) [Unverified
+                                        // Contract] [OBSOLETE]
+  PAYLOAD_CONTRACT_INVALID,    // Payload function signature and contract both
+                               // are whitelisted but doesn't match [Invalid
+                               // Transaction]
+  PAYLOAD_WHITELISTED,    // Payload is completely recognized [Clear Signing]
+} PAYLOAD_STATUS;
+
+/**
+ * @brief Struct to store Unsigned Ethereum Transaction details.
+ * @details
+ *
+ * @see
+ * @since v1.0.0
+ *
+ * @note
+ */
+#pragma pack(push, 1)
+typedef struct {
+  uint8_t nonce_size[1];
+  uint8_t nonce[32];
+
+  uint8_t gas_price_size[1];
+  uint8_t gas_price[32];
+
+  uint8_t gas_limit_size[1];
+  uint8_t gas_limit[32];
+
+  uint8_t to_address[20];
+
+  uint8_t value_size[1];
+  uint8_t value[ETH_VALUE_SIZE_BYTES];
+
+  uint64_t payload_size;
+  uint8_t *payload;
+
+  uint8_t chain_id_size[1];
+  uint8_t chain_id[8];
+
+  uint8_t dummy_r[1];
+  uint8_t dummy_s[1];
+
+  PAYLOAD_STATUS payload_status;
+} eth_unsigned_txn;
```

Comment:
Would this differ fir other EVMs?
In that case can we rename as evm_unsigned_txn?

Reply:
renamed